table.go

```diff
@@ -47,84 +59,71 @@ type RoutingTable struct {
 // NewRoutingTable creates a new routing table with a given bucketsize, local ID, and latency tolerance.
 func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerstore.Metrics) *RoutingTable {
 	rt := &RoutingTable{
-		Buckets:     []*Bucket{newBucket()},
-		bucketsize:  bucketsize,
-		local:       localID,
-		maxLatency:  latency,
-		metrics:     m,
-		PeerRemoved: func(peer.ID) {},
-		PeerAdded:   func(peer.ID) {},
+		Buckets:        []*Bucket{newBucket()},
+		bucketsize:     bucketsize,
+		local:          localID,
+		maxLatency:     latency,
+		metrics:        m,
+		cplRefreshedAt: make(map[uint]time.Time),
```

Comment:
There can be at most 16 of these. Let's just use an array: `[MaxCplForRefresh]time.Time`.

Reply:
Using a map makes allows easy "tracking" of which cpl's we have looked at so that we only refresh cpl's among those i.e. if an entry/cpl exists in the map, only then is it of interest to us.